app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,49 @@
+class Api::V1::CommentsController < Api::V1::ApplicationController
+  def index
+    @comments = Comment.where(post_id: params[:post_id])
+
+    render json: @comments, only: %i[id text post_id author_id]
+  end
+
+  def show
+    render json: @commment
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.author = User.find(params[:user_id])
+    @comment.post = Post.find(params[:post_id])
+
+    if @comment.save
+      render json: { status: 'SUCCESS', message: 'Comment was created successfully!', data: @comment }, status: :created
+    else
+      render json: @comment.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /comments/1
+  def update
+    if @comment.update(comment_params)
+      render json: @comment
+    else
+      render json: @comment.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /comments/1
+  def destroy
+    @comment.destroy
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_comment
+    @comment = Comment.find(params[:id])
+  end
+
+  # Only allow a list of trusted parameters through.
+  def comment_params
+    params.require(:comment).permit(:id, :text)
+  end
```

Comment:
- [ ] Please getting the params from the user is not working like this always the user is getting this error when he tries to post on `api/v1/users/:user_id/posts/:post_id/comments` 
![image](https://user-images.githubusercontent.com/74744787/216725659-b4acfe70-aec3-4b34-a091-49d550a38f7f.png)
 
you can fix it by just permitting the parameter from  the params directly without the require(:comments) like this
![image](https://user-images.githubusercontent.com/74744787/216725850-d3177a3e-c832-4d48-baaa-9e4cf968caa8.png)
you see it is working now

Reply:
Thanks for the review, @MeqdamAlqudah.  I will implement the change you are requiring the change required.